cirq-core/cirq/ops/boolean_hamiltonian.py

```diff
@@ -177,25 +177,25 @@ def _simplify_commuting_cnots(
 
     target, control = (0, 1) if flip_control_and_target else (1, 0)
 
-    i = 0
-    qubit_to_index: Dict[int, int] = {cnots[i][control]: i} if cnots else {}
-    for j in range(1, len(cnots)):
-        if cnots[i][target] != cnots[j][target]:
-            # The targets (resp. control) don't match, so we reset the search.
-            i = j
-            qubit_to_index = {cnots[j][control]: j}
+    to_remove = [False] * len(cnots)
```

Comment:
This can be a set of indices to remove. No need to iterate over N `False`s.

Reply:
I see. You are right.